test/distributed/test_inductor_collectives.py

```diff
@@ -23,6 +24,23 @@
 from torch._inductor.utils import has_triton, run_and_get_triton_code
 import torch._dynamo.logging
 
+
+@contextmanager
+def _dynamo_dynamic_shapes():
+    dynamic_shapes_prev_value = torch._dynamo.config.dynamic_shapes
+    capture_dynamic_output_shape_ops_prev_value = torch._dynamo.config.capture_dynamic_output_shape_ops
+    capture_scalar_outputs_prev_value = torch._dynamo.config.capture_scalar_outputs
+    torch._dynamo.config.dynamic_shapes = True
+    torch._dynamo.config.capture_dynamic_output_shape_ops = True
+    torch._dynamo.config.capture_scalar_outputs = True
```

Comment:
you don't need to do this by hand; we have torch._dynamo.config.patch(...) which can do this for you